mlir/include/mlir/Dialect/MIOpen/MIOpenOps.td

```diff
@@ -40,7 +40,12 @@ def MIOpen_Conv2DOp :
     MIOpen_Op<"conv2d">,
     Arguments<(ins MemRefRankOf<[F32, F16, BF16, I8], [5]>:$filter,
                    MemRefRankOf<[F32, F16, BF16, I8], [5]>:$input,
-                   MemRefRankOf<[F32, F16, BF16, I32], [5]>:$output)> {
+                   MemRefRankOf<[F32, F16, BF16, I32], [5]>:$output,
+                   StrAttr:$arch,
+                   I32Attr:$numCu,
+                   OptionalAttr<I32Attr>:$blockSize,
+                   OptionalAttr<I32Attr>:$gridSize,
+                   MIOpen_OptionalGemmParams:$params)> {
```

Comment:
I know you've had a conversation about @jungpark-mlir about this but I still feel this is very awkward to be as a convolution attribute. What's your plan on using them? Are you plan on them to be there for debugging purpose? It also looks like they are ignored in conv->gemm lowering.

 If they are for debugging purpose, then it is a duplication of `blocksizeoverrid` or `gridsizeoverride`. If it isn't for debugging I can't figure out how `blockSize` or `gridSize` should be derived without yet lowering to gemm yet?

Reply:
This isn't for debugging. The block size, grid size, and tuning parameters *are* used in the gridwise GEMM lowering, and so must be present for gridwise gemm. They're derived by `--affix-tuning-params`, hence them being optional attributes.

We've always been implicitly setting these attributes: this change just makes their existence explicit.

Reply:
I think I've misunderstood current vs future. Yes, right now affix tuning parameters derive all that before a conv op therefore this is all reasonable stuff.